src/SensitiveFloat.jl

```diff
@@ -0,0 +1,275 @@
+
+export multiply_sfs!, add_scaled_sfs!, combine_sfs!, add_sources_sf!
+
+@doc """
+A function value and its derivative with respect to its arguments.
+
+Attributes:
+  v:  The value
+  d:  The derivative with respect to each variable in
+      P-dimensional VariationalParams for each of S celestial objects
+      in a local_P x local_S matrix.
+  h:  The second derivative with respect to each variational parameter,
+      in the same format as d.  This is used for the full Hessian
+      with respect to all the sources.
+  hs: An array of per-source Hessians.  This will generally be reserved
+      for the Hessian of brightness values that depend only on one source.
+""" ->
+type SensitiveFloat{ParamType <: CelesteTypes.ParamSet, NumType <: Number}
+    # Actually a single value, but an Array to avoid memory allocation
+    v::Vector{NumType}
```

Comment:
While we're looking at SensitiveFloat, I wanted to ask why `v` is an array instead of just `v::Numtype` here. The effect of using an array is opposite of what the comment implies.